internal/nlwrap/nlwrap_linux.go

```diff
@@ -5,24 +5,27 @@
 // wrapped. Functions that use the package handle need to be called as "nlwrap.X"
 // instead of "netlink.X".
 //
-// The wrapped functions currently return EINTR when NLM_F_DUMP_INTR flagged
-// in a netlink response, meaning something changed during the dump so results
-// may be incomplete or inconsistent.
+// When netlink.ErrDumpInterrupted is returned, the wrapped functions retry up to
+// maxAttempts times. This error means NLM_F_DUMP_INTR was flagged in a netlink
+// response, meaning something changed during the dump so results may be
+// incomplete or inconsistent.
 //
-// At present, the possibly incomplete/inconsistent results are not returned
-// by netlink functions along with the EINTR. So, it's not possible to do
-// anything but retry. After maxAttempts the EINTR will be returned to the
-// caller.
+// To avoid retrying indefinitely, if netlink.ErrDumpInterrupted is still
+// returned after maxAttempts, the wrapped functions will discard the error, log
+// a stack trace to make the issue visible and aid in debugging, and return the
+// possibly inconsistent results. Returning possibly inconsistent results matches
+// the behaviour of vishvananda/netlink versions prior to 1.2.1, in which the
+// NLM_F_DUMP_INTR flag was ignored.
 package nlwrap
 
 import (
 	"context"
 	"errors"
 
 	"github.com/containerd/log"
+	perrors "github.com/pkg/errors"
```

Comment:
We could also swap out one for the other, instead of importing both (looks like it's only using 'errors.Is' which I think is an alias in pkg/errors)

Reply:
Done.